### PR TITLE
[new] Added support for <optgroup> in select field

### PR DESCRIPTION
**Pull request by mattwatsoncodes**
Added support for <optgroup> in the select field if an option is passed through as an array. 

Screenshot 2021-08-10 at 16 03 14

### CODE REVIEW

File: wp-settings-framework.php [hunk 1]
```diff
@@ -536,6 +536,16 @@ public function generate_select_field( $args ) {
 			echo '<select name="' . $args['name'] . '" id="' . $args['id'] . '" class="' . $args['class'] . '">';
 
 			foreach ( $args['choices'] as $value => $text ) {
+				if ( is_array( $text ) ) {
+					echo sprintf( '<optgroup label="%s">', $value );
```
Comment (pramodjodhani) [inline wp-settings-framework.php:540]: Maybe we should use esc_attr( $value  ) here?

File: wp-settings-framework.php [hunk 2]
```diff
@@ -536,6 +536,16 @@ public function generate_select_field( $args ) {
...
+					foreach ( $text as $group_value => $group_text ) {
+						$selected = $group_value == $args['value'] ? 'selected="selected"' : '';
+						echo sprintf( '<option value="%s" %s>%s</option>', $group_value, $selected, $group_text );
```
Comment (pramodjodhani) [inline wp-settings-framework.php:543]: We could use esc_* functions here too

**Review by pramodjodhani — CHANGES_REQUESTED** [review]

**Comment by mattwatsoncodes** [comment]
Sorted @pramodjodhani and escaped a few other parts of that function while I was there too :)

**Comment by pramodjodhani** [comment]
Hi @mattwatsoncodes,  Yes it looks great. You haven't re-requested the review but consider it approved from my end 🙌

**Pull request merged**